tests/cbor_encode/test3_strange/src/main.c

```diff
@@ -696,6 +696,122 @@ void test_range(void)
 	zassert_mem_equal(exp_payload_range3, output, sizeof(exp_payload_range3), NULL);
 }
 
+
+void test_value_range(void)
+{
+	const uint8_t exp_payload_value_range1[] = {LIST(6),
+		11,
+		0x19, 0x03, 0xe7, // 999
+		0x29, // -10
+		1,
+		0x18, 42, // 42
+		0x65, 0x77, 0x6f, 0x72, 0x6c, 0x64, // "world"
+		END
+	};
+
+	const uint8_t exp_payload_value_range2[] = {LIST(6),
+		0x18, 100, // 100
+		0x39, 0x03, 0xe8, // -1001
+		0x18, 100, // 100
+		0,
+		0x18, 42, // 42
+		0x65, 0x77, 0x6f, 0x72, 0x6c, 0x64, // "world"
+		END
+	};
+
+	struct ValueRange input1 = {
+		._ValueRange_greater10 = 11,
+		._ValueRange_less1000 = 999,
+		._ValueRange_greatereqmin10 = -10,
+		._ValueRange_lesseq1 = 1,
+	};
+	struct ValueRange input2 = {
+		._ValueRange_greater10 = 100,
+		._ValueRange_less1000 = -1001,
+		._ValueRange_greatereqmin10 = 100,
+		._ValueRange_lesseq1 = 0,
+	};
+	struct ValueRange input3_inval = {
+		._ValueRange_greater10 = 10,
+		._ValueRange_less1000 = 999,
+		._ValueRange_greatereqmin10 = -10,
+		._ValueRange_lesseq1 = 1,
+	};
+	struct ValueRange input4_inval = {
+		._ValueRange_greater10 = 11,
+		._ValueRange_less1000 = 1000,
+		._ValueRange_greatereqmin10 = -10,
+		._ValueRange_lesseq1 = 1,
+	};
+	struct ValueRange input5_inval = {
+		._ValueRange_greater10 = 11,
+		._ValueRange_less1000 = 999,
+		._ValueRange_greatereqmin10 = -11,
+		._ValueRange_lesseq1 = 1,
+	};
+	struct ValueRange input6_inval = {
+		._ValueRange_greater10 = 11,
+		._ValueRange_less1000 = 999,
+		._ValueRange_greatereqmin10 = -10,
+		._ValueRange_lesseq1 = 2,
+	};
+	struct ValueRange input7_inval = {
+		._ValueRange_greater10 = 1,
+		._ValueRange_less1000 = 999,
+		._ValueRange_greatereqmin10 = -10,
+		._ValueRange_lesseq1 = 1,
+	};
+	struct ValueRange input8_inval = {
+		._ValueRange_greater10 = 11,
+		._ValueRange_less1000 = 10000,
```

Comment:
I'm a bit confused about these structs and why some numbers aren't correct against their greater/less value. This is `999` and `1000` in some other structs, so possible extra zero typo?

Reply:
Some of these are invalid tests (`*_inval`), which tests that the code fails on those.